### PR TITLE
feat(math): adding Dec.SdkIntTrim

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

Closes: #987

Benchmarks:

```
goos: linux
goarch: amd64
BenchmarkSdkIntTrim/exp-16               1000000              1245 ns/op
BenchmarkSdkIntTrim/quo-integer-16        408232              2490 ns/op
BenchmarkSdkIntTrim/string-16             598272              2685 ns/op
```

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: types/math/dec_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func BenchmarkSdkIntTrim(b *testing.B) {
+	s := "12345678901234567890.12345678901234567890"
+	d, err := NewDecFromString(s)
+	if err != nil {
+		b.Error("can't converet test number")
```
Comment (ryanchristo) [inline types/math/dec_bench_test.go:13]:
```suggestion
		b.Error("can't convert test number")
```

**Review by ryanchristo — COMMENTED** [review]
This looks good to me.

You previously mentioned this was something you would like to see up-streamed to the sdk and said you would prefer @amaurym and @anilCSE to review. Might be good to get a review from @aaronc as well.

**Review by technicallyty — APPROVED** [review]
LGTM. Not sure what will get merged first, so might be nice to create an issue to update buy_direct with this utility function

File: types/math/dec_bench_test.go (hunk 1 above)
Comment (robert-zaremba) [reply types/math/dec_bench_test.go:13]: In SDK we don't use apd.

**Comment by robert-zaremba** [comment]
> You previously mentioned this was something you would like to see up-streamed to the sdk

The whole library. In SDK we don't use apd.

**Review by aaronc — APPROVED** [review]

**Comment by aaronc** [comment]
We should be thinking about upstreaming this `math` package to the SDK (as a `math` go module) in the near future. It has long been known that the current SDK `Dec` is not ideal (https://github.com/cosmos/cosmos-sdk/issues/7773). Either apd or https://github.com/EricLagergren/decimal should be the basis of what we use in the SDK in the future. If anyone is able to compare those two libraries btw that we be great so we can make an informed decision and best support things upstream. See also https://github.com/cockroachdb/apd/issues/100

**Pull request merged**